test/contrib/hsgp/test_approximation.py

```diff
@@ -100,13 +127,26 @@ def test_kernel_approx_squared_exponential(
     assert x1.shape == x2.shape
     m = 100  # large enough to ensure the approximation is accurate
     dim = x1.shape[-1]
+    if xfail:
+        with pytest.raises(ValueError):
+            diag_spectral_density_squared_exponential(1.0, length, ell, m, dim)
+        return
     spd = diag_spectral_density_squared_exponential(1.0, length, ell, m, dim)
 
     eig_f1 = eigenfunctions(x1, ell=ell, m=m)
     eig_f2 = eigenfunctions(x2, ell=ell, m=m)
     approx = (eig_f1 * eig_f2) @ spd
-    exact = RBF(length)(x1, x2)
-    assert jnp.isclose(approx, exact, rtol=1e-3)
+
+    def _exact_rbf(length):
+        return RBF(length)(x1, x2).squeeze(axis=-1)
+
+    if isinstance(length, float | int):
```

Comment:
The tests are failing becuase of this line

``python
>       if isinstance(length, float | int):
E       TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

Maybe you can use

```python
 isinstance(length, float) |  isinstance(length, int)
```
I am not sure if there is a more elegant way

Reply:
Looks like the union shorthand is only for Python 3.10 and higher. We could just use the `Union` constructor explicitly, which should support anything above 3.5.